src/core/Akka.Remote.TestKit/BarrierCoordinator.cs

```diff
@@ -299,19 +290,19 @@ public override int GetHashCode()
 
         public sealed class WrongBarrierException : Exception
         {
-            public WrongBarrierException(string barrier, IActorRef client, Data barrierData)
-                : base($"[{client}] tried to enter '{barrier}' while we were waiting for '{barrierData.Barrier}'")
+            public WrongBarrierException(string barrier, IActorRef client, RoleName roleName, Data barrierData)
+                : base($"[{client}] [{roleName}] tried to enter '{barrier}' while we were waiting for '{barrierData.Barrier}'")
```

Comment:
Expanded the description here of WHICH ROLE tried to enter the wrong barrier - this should make it much easier to debug tests in the future that have issues with nodes crossing barriers at inappropriate times.